src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,9 +1,219 @@
-<script setup>
-</script>
 <template>
-    <div class="ma-keywords-generator">
-        <div class="ma-header">
+  <div class="ma-keywords-generator">
+    <div class="card">
+      <div class="container">
+        <div :class="['input-container', { 'centered': isKeywordsGenerated }]">
+          <div class="ma-header">
             <span>Keyword Generator</span>
+          </div>
+          <textarea ref="textarea" v-model="text" placeholder="Enter text here..." rows="6" @input="adjustTextareaHeight"></textarea>
+          <button @click="generateKeywords">Generate Keywords</button>
+        </div>
+        <div v-if="keywords.unigrams.length || keywords.bigrams.length || keywords.trigrams.length" class="keywords-output">
+          <div class="ma-header">
+            <span>Generated Keywords</span>
+          </div>
+          <div class="filter-container">
+            <label for="ngram-filter">Filter by:</label>
+            <select id="ngram-filter" v-model="selectedNGram" @change="filterKeywords">
+              <option value="unigrams">1-grams</option>
+              <option value="bigrams">2-grams</option>
+              <option value="trigrams">3-grams</option>
+            </select>
+          </div>
+          <div>
+            <h4>{{ getNGramLabel() }}</h4>
+            <ul>
+              <li v-for="(item, index) in filteredKeywords" :key="index">{{ item }}</li>
+            </ul>
+          </div>
         </div>
+      </div>
     </div>
-</template>
\ No newline at end of file
+  </div>
+</template>
+
+<script setup>
+import { ref, computed, onMounted } from 'vue';
+
+const text = ref('');
+const keywords = ref({
+  unigrams: [],
+  bigrams: [],
+  trigrams: []
+});
```

Comment:
The `n-gram` structure should be dynamic. Later, when the n-gram count increases/decreases, you will not need to update these parts.